keymapper/migrations.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# key-mapper - GUI for device specific keyboard mappings
+# Copyright (C) 2021 sezanzeb <proxima@sezanzeb.de>
+#
+# This file is part of key-mapper.
+#
+# key-mapper is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# key-mapper is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with key-mapper.  If not, see <https://www.gnu.org/licenses/>.
+
+
+"""Migration functions"""
+
+
+import os
+import json
+import copy
+
+from pathlib import Path
+
+from keymapper.logger import logger
+from keymapper.paths import get_preset_path, mkdir, CONFIG_PATH
+
+
+def all_presets():
+    p = Path(get_preset_path())
+    presets = []
+    for folder in p.iterdir():
+        if folder.is_dir():
+            for preset in folder.iterdir():
+                if preset.suffix == ".json":
+                    presets.append(preset)
```

Comment:
https://medium.com/swlh/return-early-pattern-3d18a41bba8

https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

```suggestion
    for folder in p.iterdir():
        if not folder.is_dir():
            continue
                
        for preset in folder.iterdir():
            if preset.suffix == ".json":
                presets.append(preset)
```